store-hints.py

```diff
@@ -27,11 +31,20 @@
         con.commit()
     for idx, (score, neighbor) in enumerate(neighbors):
         con.execute ("insert into nearby (word, neighbor, similarity, percentile) values (?, ?, ?, ?)", (secret, neighbor, "%s" % score, (1 + idx)))
-
-    top = neighbors[-2][0]
-    top10 = neighbors[-12][0]
-    rest = neighbors[0][0]
-    con.execute ("insert into similarity_range (word, top, top10, rest) values (?, ?, ?, ?)", (secret, "%s" % top, "%s" % top10, "%s" % rest))
+    try:
+        top = neighbors[-2][0]
+        top10 = neighbors[-12][0]
+        rest = neighbors[0][0]
+        con.execute ("insert into similarity_range (word, top, top10, rest) values (?, ?, ?, ?)", (secret, "%s" % top, "%s" % top10, "%s" % rest))
+    except:
```

Comment:
Catch index out of range here. If the con.execute line excepts then you are sending the wrong message.